lib/cronofy/auth.rb

```diff
@@ -3,18 +3,19 @@
 module Cronofy
   class Auth
     API_URL = 'https://api.cronofy.com'
-    APP_URL = 'https://app.cronofy.com'
 
-    def initialize(client_id, client_secret, token)
+    attr_reader :access_token
+
+    def initialize(client_id, client_secret, token, refresh_token=nil)
       @client = OAuth2::Client.new(client_id, client_secret, site: API_URL)
-      @token = token
+      @access_token = OAuth2::AccessToken.new(@client, token, { refresh_token: refresh_token })
     end
 
     def user_auth_link(redirect_uri)
       url = @client.auth_code.authorize_url(
         :redirect_uri => redirect_uri,
         :response_type => 'code'
-      ).gsub(API_URL, APP_URL)
+      )
```

Comment:
Actually, I think the OAuth gem will deal with it for you from what I remember of reading the source so this might work:

```ruby
@auth_client.auth_code.authorize_url(
  :redirect_uri => redirect_uri,
  :response_type => 'code',
  :scope => scope.join(' '),
)
```